### DISCUSSION TITLE
I have max-duration problem on python flask deployment

### DISCUSSION THREAD

**Discussion by JINYUHOON** [vercel/community#6562] [Help]
### Summary

python flask max-duration trouble

### Steps to Reproduce

I am distributing my Python flask project on vercel.
However, I encountered a max-duration issue based on the API request and modified the vercel.json file to solve it. But this problem still persists.
I'm making a request from a js file in my root. In this case, how should I configure it to work correctly?

my vercel.json is

```
{
    "version" : 2,
    "builds": [
        {
            "src": "application.py",
            "use": "@vercel/python",
            "config": { "maxLambdaSize": "15mb", "maxDuration": 300, "runtime": "python3.9" }
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "application.py"
        }
    ]
  }
```

**Comment by versecafe** [comment]
@JINYUHOON what is the actual issue? is it killing the function before the specified maximum or are you trying to go past the 300 second max?

**Reply by JINYUHOON** [reply]
We just want to adjust maxDuration . I'm trying to set it to 300 seconds.

**Reply by versecafe** [reply]
Are you on a Hobby or Pro/Enterprise team for the project?

**Reply by JINYUHOON** [reply]
Yes , I'm using pro plan
For projects executed with next.js, max Duration was set to 300 seconds.
In a project running with flask, you want to set maxduration in vercel.json, including the build above.

**Reply by versecafe** [reply]
```json
{
  "version": 2,
  "rewrites": [{ "source": "/(.*)", "destination": "/application" }],
  "functions": {
    "application.py": {
      "memory": 128,
      "maxDuration": 300
    }
  }
}
```
this config should work for you and for pro the duration cap is 300, and enterprise 900, and you can adjust the memory between 128 and 1024 MB

**Reply by JINYUHOON** [reply]
```
Error: The pattern "application.py" defined in `functions` doesn't match any Serverless Functions inside the `api` directory.
```
When I change the settings with the provided solution, I encounter the above error.

[a comment by an automated account is not shown]
